Arcade abs sync

techeyy opened the pull request:
Players using Arcade ABS will no longer appear as if they are locking their tires and creating laggy smoke trails behind them. 

https://www.youtube.com/watch?v=4-NsVA7Zup0

techeyy committed:
Adds support for Arcade ABS assist

techeyy committed:
Update to development version

OfficialLambdax commented on lua/ge/extensions/MPVehicleGE.lua:
```diff
@@ -1324,11 +1328,13 @@ local function applyVehSpawn(event)
 		spawnedVeh = spawn.spawnVehicle(vehicleName, serialize(vehicleConfig), pos, rot, { autoEnterVehicle=false, vehicleName="multiplayerVehicle", cling=true})
 		spawnedVehID = spawnedVeh:getID()
 		spawnedVeh:setField("protected", 0, protected or "0")
+		spawnedVeh:setField("absMode", 0, absMode or "realistic")
```
Unlikely required to set a new field here

OfficialLambdax commented on lua/ge/extensions/MPVehicleGE.lua:
```diff
@@ -1422,6 +1430,7 @@ local function applyVehEdit(serverID, data)
 	end
 	
 	veh:setField("protected", 0, protected or "0")
+	veh:setField("abs", 0, absMode or "realistic")
```
Field is called `absMode` not `abs`

OfficialLambdax reviewed (dismissed):
Field set seems unnecessary. Otherwise looks good. Untested

techeyy committed:
arcade ABS fix

fixed line 1433 typo, however still using vehicle field to store ABS mode

techeyy committed:
Merge branch 'BeamMP:development' into Arcade-ABS-Sync

techeyy committed:
Merge branch 'development' into Arcade-ABS-Sync

techeyy committed:
Add files via upload

techeyy committed:
Add files via upload

ABS behavior now applied on vehicle reset, no longer "forgets" ABS behavior when CTRL+R is used by the observing player.

Note that when a driver changes their assist setting, proper behavior is only guaranteed for the observer once they queue the driver's change. (non issue)

techeyy committed:
arcade ABS sync

techeyy committed:
Merge branch 'development' into Arcade-ABS-Sync
